app/code/Magento/SalesRule/Plugin/CouponUsagesIncrement.php

```diff
@@ -34,14 +36,16 @@ public function __construct(
      * Increments number of coupon usages after placing order.
      *
      * @param OrderService $subject
-     * @param OrderInterface $result
-     * @return OrderInterface
+     * @param OrderInterface $order
+     * @return array
+     * @throws CouponUsageExceeded
+     * @throws LocalizedException
      * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
```

Comment:
this plugin should happen before order place, to avoid order saving in case of coupon validation fail